torchdiffeq/_impl/misc.py

```diff
@@ -176,6 +177,28 @@ def forward(self, t, y):
         return self.mul * self.base_func(-t, y)
 
 
+class _PerturbFunc(torch.nn.Module):
+    _inf = torch.tensor(math.inf)
+    _neginf = torch.tensor(-math.inf)
+
+    def __init__(self, base_func):
+        super(_PerturbFunc, self).__init__()
+        self.base_func = base_func
+
+    def forward(self, t, y, *, perturb=None):
+        # This dtype change here might be buggy.
+        # The exact time value should be determined inside the solver,
+        # but this can slightly change it due to numerical differences during casting.
```

Comment:
I can believe that something could go wrong here, but I'm not sure what -- do you have an explicit example or is this just a general concern?

Reply:
Nope, no explicit example. It should be okay to leave it like this until we actually have to deal with it.